livesync_citadel/indico_livesync_citadel/schemas.py

```diff
@@ -0,0 +1,266 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import itertools
+
+from marshmallow import post_dump
+from tika import parser
+from webargs import fields
+
+from indico.core.db.sqlalchemy.links import LinkType
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.core.db.sqlalchemy.protection import ProtectionMode
+from indico.core.marshmallow import mm
+from indico.modules.attachments.models.attachments import Attachment, AttachmentType
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.notes.models.notes import EventNote
+from indico.web.flask.util import url_for
+
+
+def _get_location(obj):
+    if obj.venue_name and obj.room_name:
+        return '{}: {}'.format(obj.venue_name, obj.room_name)
+    elif obj.venue_name or obj.room_name:
+        return obj.venue_name or obj.room_name
+    else:
+        return ''
+
+
+def _get_identifiers(principal):
+    if principal.principal_type == PrincipalType.user:
+        # Instead of using the email this uses `User:ID`.
+        # Since the user can change the email this is better as
+        # it will ensure that only this given Indico user has access.
+        # If you want to stick with email, simply replace it with
+        # 'User:{}'.format(principal.email)
+        yield principal.identifier
+        yield principal.email
+    elif principal.principal_type == PrincipalType.event_role:
```

Comment:
Category roles are very similar, so we can handle those here as well

Reply:
Actually, should we descend into either of the roles or just add their identifier? I don't think updating the search data happens in case role membership changes... and at least for category roles doing so would not really be practical